src/txmempool.cpp

```diff
@@ -136,14 +136,15 @@ void CTxMemPool::UpdateTransactionsFromBlock(const std::vector<uint256> &vHashes
         // include them, and update their setMemPoolParents to include this tx.
         for (; iter != mapNextTx.end() && iter->first->hash == hash; ++iter) {
             const uint256 &childHash = iter->second->GetHash();
+            // We can skip updating entries that are in the block (which are
+            // already accounted for).
+            if (setAlreadyIncluded.count(childHash)) continue;
```

Comment:
one effect of adding this here would be that `setChildren.insert` won't be called for those that are skipped here anymore, but since `setChildren` is locally declared and only used once below, it looks fine.

This change lgtm since it's only adding a check to be called earlier to avoid an unnecessary `mapTx.find()` call and the rest of the boolean looks the same.